ibft/msg_processing.go

```diff
@@ -38,20 +38,14 @@ func (i *Instance) ProcessChangeRoundPartialQuorum() (found bool, err error) {
 	if msgsMap := i.MsgQueue.MessagesForIndex(msgqueue.IBFTAllRoundChangeIndexKey(i.State.Lambda, i.State.SeqNumber)); msgsMap != nil {
 		// get values and keys slices
 		msgs := make([]*network.Message, 0)
-		msgsIds := make([]string, 0)
-		for id, msg := range msgsMap {
+		for _, msg := range msgsMap {
 			msgs = append(msgs, msg)
-			msgsIds = append(msgsIds, id)
 		}
 
 		found, err := i.uponChangeRoundPartialQuorum(msgs)
 		if err != nil {
 			return false, err
 		}
-		if found {
```

Comment:
Could it lead to messages that are never deleted?

Reply:
no as we purge the queue for the index when an instance decides